tests/unit/matrix/test_cubemeasure.py

```diff
@@ -1316,10 +1345,22 @@ def it_knows_its_weighted_counts(self, raw_weighted_counts):
             )
         )
 
+    def it_knows_its_zscores(self, raw_weighted_counts):
+        weighted_cube_counts = _CatXMrWeightedCubeCounts(None, raw_weighted_counts)
+        assert weighted_cube_counts.zscores.tolist() == [
+            [-1.7549928774784245, -1.6818357317441646, -1.754992877478425],
+            [1.7549928774784245, 1.6818357317441637, 1.754992877478425],
+        ]
+
+    def it_does_not_calculate_zscores_when_matrix_is_defective(self):
+        raw_weighted_counts = np.array([[[1, 1], [2, 2]]])
+        weighted_cube_counts = _CatXMrWeightedCubeCounts(None, raw_weighted_counts)
+        assert weighted_cube_counts.zscores.tolist() == [[0, 0]]
```

Comment:
Why are these zeroes instead of NaNs if this array is defective? A 2 x 2 array of integer values is not defective. Also, these are weighted counts and so are floats; better to use 1.1, 2.2 etc. rather than integers. It's just as easy to get it right and more closely approximates the runtime values.

Reply:
it is defective for a catxmr matrix
```
        if not np.all(counts.shape) or np.linalg.matrix_rank(counts) < 2:
            return np.zeros(counts.shape)
```